src/test/util/setup_common.cpp

```diff
@@ -91,6 +93,16 @@ void Seed(FastRandomContext& ctx)
     ctx = FastRandomContext(seed);
 }
 
+CAmount SemiRandMoneyAmount()
+{
+    switch (InsecureRandRange(100)) {
+    case 0: return 0;
+    case 1: return 1;
+    case 2: return MAX_MONEY;
+    default: return InsecureRandRange(InsecureRandBool() ? COIN : MAX_MONEY);
```

Comment:
Can you explain the motivation for restricting the range to `COIN` half the time? Is it just because they're more common values?

Reply:
Thanks for looking! Yes, the idea was to often generate a real-life value, while still testing the full `MoneyRange` in `src/consensus/amount.h`.

Reply:
Updated its Doxygen doc with that info.